build_test.go

```diff
@@ -86,8 +88,40 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 		Expect(publishProcess.ExecuteCall.Receives.ProjectPath).To(Equal("some/project/path"))
 		Expect(publishProcess.ExecuteCall.Receives.OutputPath).To(MatchRegexp(`dotnet-publish-output\d+`))
 
-		Expect(buffer.String()).To(ContainSubstring("Some Buildpack some-version"))
+		Expect(buffer.String()).To(ContainSubstring("Some Buildpack 0.0.1"))
 		Expect(buffer.String()).To(ContainSubstring("Executing build process"))
+		Expect(buffer.String()).To(ContainSubstring("WARNING: Setting the project path through buildpack.yml will be deprecated soon in Dotnet Publish Buildpack v1.0.0"))
+		Expect(buffer.String()).To(ContainSubstring("Please specify the project path through the $BP_DOTNET_PROJECT_PATH environment variable instead. See README.md or the documentation on paketo.io for more information."))
+	})
+
+	context("when project path is set via BP_DOTNET_PROJECT_PATH", func() {
+		it.Before(func() {
+			Expect(os.Setenv("BP_DOTNET_PROJECT_PATH", "some/project/path"))
+		})
+
+		it("returns a build result", func() {
+			result, err := build(packit.BuildContext{
+				WorkingDir: workingDir,
+				BuildpackInfo: packit.BuildpackInfo{
+					Name:    "Some Buildpack",
+					Version: "some-version",
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(Equal(packit.BuildResult{}))
+
+			Expect(sourceRemover.RemoveCall.Receives.WorkingDir).To(Equal(workingDir))
+			Expect(sourceRemover.RemoveCall.Receives.PublishOutputDir).To(MatchRegexp(`dotnet-publish-output\d+`))
+			Expect(sourceRemover.RemoveCall.Receives.ExcludedFiles).To(ConsistOf([]string{".dotnet_root"}))
+
+			Expect(publishProcess.ExecuteCall.Receives.WorkingDir).To(Equal(workingDir))
+			Expect(publishProcess.ExecuteCall.Receives.RootDir).To(Equal("some-existing-root-dir"))
+			Expect(publishProcess.ExecuteCall.Receives.ProjectPath).To(Equal("some/project/path"))
+			Expect(publishProcess.ExecuteCall.Receives.OutputPath).To(MatchRegexp(`dotnet-publish-output\d+`))
+
+			Expect(buffer.String()).To(ContainSubstring("Some Buildpack some-version"))
+			Expect(buffer.String()).To(ContainSubstring("Executing build process"))
+		})
 	})
```

Comment:
```suggestion
	}, spec.Sequential())
```
I don't think that setting env var is thread safe for test so I would recommend making the test in which you do so run sequentially.

Reply:
In the init test, we don't run these tests in parallel (we use the default sequential mode). Do you think this is still relevant to include?

Reply:
+1 Frankie, I was also thinking that if two diff test suites like `build_test.go` and `detect_test.go` both were individually sequential, but parallel in the init test, there's a good chance that their env vars could still affect each other

Reply:
If that is the case I retract my previous statement.